src/main/webapp/app/shared/notification/connection-notification/connection-notification.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import { AccountService } from 'app/core/auth/account.service';
+import { JhiWebsocketService } from 'app/core/websocket/websocket.service';
+import { User } from 'app/core/user/user.model';
+import { Alert, AlertService } from 'app/core/util/alert.service';
+
+@Injectable({ providedIn: 'root' })
+export class ConnectionNotificationService {
+    alert?: Alert;
+    connected?: boolean;
+
+    constructor(private accountService: AccountService, private jhiWebsocketService: JhiWebsocketService, private alertService: AlertService) {}
+
+    init() {
+        this.accountService.getAuthenticationState().subscribe((user: User | undefined) => {
+            if (user) {
+                // listen to connect / disconnect events
+                this.jhiWebsocketService.enableReconnect();
+                this.jhiWebsocketService.bind('connect', this.onConnect);
+                this.jhiWebsocketService.bind('disconnect', this.onDisconnect);
+            } else {
+                // On logout, reset component
+                this.connected = undefined;
+                this.alert?.close();
+                this.alert = undefined;
+                this.jhiWebsocketService.disableReconnect();
+                this.jhiWebsocketService.unbind('connect', this.onConnect);
+                this.jhiWebsocketService.unbind('disconnect', this.onDisconnect);
+            }
+        });
+    }
+
+    /**
+     * Only update on connect if there is not already an active connection.
+     * This alert is temporary and disappears after 10 seconds.
+     **/
+
+    onConnect = () => {
+        if (this.connected === false) {
+            this.alert?.close();
+            this.alert = this.alertService.addAlert({ type: 'success', message: 'artemisApp.connectionAlert.reconnected', timeout: 10000 });
+        }
+        this.connected = true;
```

Comment:
That's actually intended as on app startup, the value of ``this.connected`` is ``undefined``; then, when the WS connects for the first time, ``this.connected`` wouldn't be set to true if we moved it into the conditional.
Setting ``this.connected`` to false as initial value is not a good idea as well as it would show a "You just reconnected" alert on each app startup; setting it to true as initial value will cause both a disconnnect and a reconect alert to show on startup as for some reason the websocket service triggers a disconnect before connecting. It was like this before this PR as well when this file was a component and not a service.